mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -63,7 +63,7 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
 
   /** @group setParam */
   @Since("1.4.0")
-  def setSplits(value: Array[Double]): this.type = set(splits, value)
+  def setSplits(value: Array[Double]): this.type = set(splits, value.filter(!_.isNaN))
```

Comment:
Yeah, that seems like an error. NaN has no ordering so it doesn't make sense to consider or return them as part of quantiles. I think these other classes have to handle this case too and they can be rejected here. It's not something this class can handle for them because it doesn't know what the quantiles would have been without NaN.